web/src/settings_ui.ts

```diff
@@ -106,3 +107,46 @@ export function disable_sub_setting_onchange(
         }
     }
 }
+
+export function set_expires_on_text(
+    $elem: JQuery,
+    custom_input_time_value: number,
+    valid_to_text: string,
+): void {
+    if ($elem.val() === "custom") {
+        $elem.parent().find(".input-time-valid-till").hide();
+        set_custom_expires_on_text($elem, custom_input_time_value, valid_to_text);
+    } else {
+        $elem.parent().find(".input-time-valid-till").show();
+        $elem.parent().find(".input-time-valid-till").text(valid_to_text);
+    }
+}
+
+export function set_custom_expires_on_text(
+    $elem: JQuery,
+    custom_input_time_value: number,
+    valid_to_text: string,
+): void {
+    if (util.validate_custom_time_input(custom_input_time_value)) {
+        $elem.parent().find(".custom-input-time-valid-till").text(valid_to_text);
+    } else {
+        $elem
+            .parent()
+            .find(".custom-input-time-valid-till")
+            .text($t({defaultMessage: "Invalid custom time"}));
+    }
+}
+
+export function set_custom_time_inputs_visibility(
+    $elem: JQuery,
+    custom_input_time_unit: string,
+    custom_input_time_value: number,
+): void {
+    if ($elem.val() === "custom") {
+        $elem.parent().find(".custom-input-time-value").val(custom_input_time_value);
+        $elem.parent().find(".custom-input-time-unit").val(custom_input_time_unit);
+        $elem.parent().find(".custom-time-input-container").show();
+    } else {
+        $elem.parent().find(".custom-time-input-container").hide();
+    }
+}
```

Comment:
Would settings_components.ts be a more appropriate home for these functions?

Reply:
Shifted them to `settings_components.ts` as discussed in [CZO](https://chat.zulip.org/#narrow/channel/6-frontend/topic/realm.3A.20Schedule.20realm.20deletion.20time.20while.20deactivating.20realm).